exonum/src/api/error.rs

```diff
@@ -16,10 +16,118 @@
 
 use failure::Fail;
 use serde::Serialize;
+
+use std::collections::HashMap;
 use std::io;
 
 use crate::node::SendError;
 
+/// List of possible HTTP response codes.
+#[derive(Debug, Serialize)]
+pub enum HttpCode {
+    #[allow(missing_docs)]
+    BadRequest = 400,
+    #[allow(missing_docs)]
+    NotFound = 404,
+    #[allow(missing_docs)]
+    NotImplemented = 501,
+}
+
+/// API HTTP error struct.
+#[derive(Fail, Debug, Serialize)]
+pub struct ApiError {
+    /// HTTP error code.
+    #[serde(skip)]
+    pub http_code: HttpCode,
+    /// A URI reference to the documentation or possible solutions for the problem.
+    #[serde(rename = "type")]
+    pub error_type: String,
+    /// Short description of the error.
+    pub title: String,
+    /// Detailed description of the error.
+    pub detail: String,
+    /// HashMap with additional parameters.
+    pub params: HashMap<String, String>,
+    /// Source of the error.
+    pub source: String,
+    /// Internal error code.
+    pub error_code: i8,
+}
+
+impl std::fmt::Display for ApiError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&serde_json::to_string(self).unwrap())
+    }
+}
+
+impl ApiError {
+    fn default() -> Self {
```

Comment:
Probably it's better to rename this method to `new` and make it take `http_code` as an argument.

Reply:
...and then it could make sense to make this method public, so we don't have to bother ourselves with creating a constructor method for every possible HTTP response code.